src/Catalyst.Core.Lib.Tests/UnitTests/P2P/Protocols/PeerDeltaHistoryRequestTest.cs

```diff
@@ -0,0 +1,149 @@
+                _testSettings,
+                _cancellationProvider
+            );
+        }
+
+        [Fact]
+        public async Task Can_Query_Expected_Peer()
+        {
+            var recipientPeerId = PeerIdHelper.GetPeerId();
+            await _peerDeltaHistoryRequest.DeltaHistoryAsync(recipientPeerId).ConfigureAwait(false);
+            var expectedDto = Substitute.For<IMessageDto<ProtocolMessage>>();
+            expectedDto.RecipientPeerIdentifier.Returns(recipientPeerId);
+            expectedDto.SenderPeerIdentifier.Returns(_testSettings.PeerId);
+            _peerDeltaHistoryRequest.PeerClient.ReceivedWithAnyArgs(1).SendMessage(Arg.Is(expectedDto));
+        }
+
+        [Fact]
+        public async Task Can_Receive_Query_Response_On_Observer()
+        {
+            var recipientPeerId = PeerIdHelper.GetPeerId();
+            
+            var hp = new HashProvider(HashingAlgorithm.GetAlgorithmMetadata("blake2b-256"));
+            var lastDeltaHash = hp.ComputeMultiHash(ByteUtil.GenerateRandomByteArray(32));
+            
+            var collection = new List<DeltaIndex>();
+
+            //// this matches the fake mock 
+            for (uint x = 0; x < 10; x++)
+            {
+                var delta = new Delta
+                {
+                    PreviousDeltaDfsHash = lastDeltaHash.Digest.ToByteString()
+                };
+
+                var index = new DeltaIndex
+                {
+                    Height = 10,
+                    Cid = delta.ToByteString()
+                };
+
+                collection.Add(index);
+                lastDeltaHash = hp.ComputeMultiHash(ByteUtil.GenerateRandomByteArray(32));
+            }
+            
+            var deltaHistoryResponse = new PeerDeltaHistoryResponse(recipientPeerId, collection);
+
+            _peerDeltaHistoryRequest.DeltaHistoryResponseMessageStreamer.OnNext(deltaHistoryResponse);
+            var response = await _peerDeltaHistoryRequest.DeltaHistoryAsync(recipientPeerId).ConfigureAwait(false);
+            response.DeltaCid.Count.Should().Be(10);
+        }
+
+        [Fact]
+        public async Task No_Response_Timeout_And_Returns_False()
+        {
+            var recipientPeerId = PeerIdHelper.GetPeerId();
+            _cancellationProvider.CancellationTokenSource.Cancel();
+            var response = await _peerDeltaHistoryRequest.DeltaHistoryAsync(recipientPeerId).ConfigureAwait(false);
+            response.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task Exception_During_Query_Returns_Null()
+        {
+            var recipientPeerId = PeerIdHelper.GetPeerId();
+            _cancellationProvider.Dispose(); //do summet nasty to force exception
+            var response = await _peerDeltaHistoryRequest.DeltaHistoryAsync(recipientPeerId).ConfigureAwait(false);
+            response.Should().BeNull();   
+        }
+
+        // [Fact]
+        // public async Task Can_Dispose_Class()
+        // {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4600656)